website/static/js/tests/addons/folderPickerNodeConfig.test.js

```diff
@@ -0,0 +1,450 @@
+                vm.fetchFromServer()
+                    .always(function(resp) {
+                        assert.deepEqual(resp, data);
+                        done();
+                    });
+            });
+        });
+        describe('#submitSettings', () => {
+            var data = makeFakeData();
+            data.urls.view = faker.internet.ip();
+            var configUrl = faker.internet.ip();
+            var endpoints = [{
+                method: 'PUT',
+                url: configUrl,
+                response: {
+                    result: data
+                }
+            }];
+            var server;
+            var spy;
+            before(() => {
+                server = utils.createServer(sinon, endpoints);
+                spy = sinon.spy($osf, 'putJSON');
+            });
+            after(() => {
+                server.restore();
+                $osf.putJSON.restore();
+            });
+            data.urls.config = configUrl;
+            it('serializes the VM state and sends a PUT request to the \'config\' url passed in settings', (done) => {
+                hardReset();
+                vm.updateFromData(data)
+                    .always(function() {
+                        vm.submitSettings()
+                            .always(function() {
+                                assert.isTrue(
+                                    spy.calledWith(data.urls.config, vm.folder().name.toUpperCase())
+                                );
+                                done();
+                            });
+                    });
+            });
+        });
+        describe('#_importAuthConfirm', () => {
+            var importAuthUrl = faker.internet.ip();
+            var endpoints = [{
+                method: 'PUT',
+                url: importAuthUrl,
+                response: {}
+            }];
+            var server;
+            var putJSONSpy;
+            var activatePickerSpy;
+            before(() => {
+                hardReset();
+                server = utils.createServer(sinon, endpoints);
+                putJSONSpy = sinon.spy($osf, 'putJSON');
+                activatePickerSpy = sinon.spy(vm, 'activatePicker');
+            });
+            after(() => {
+                server.restore();
+                $osf.putJSON.restore();
+                activatePickerSpy.restore();
+            });
+            var data = makeFakeData();
+            data.urls.importAuth = importAuthUrl;
+            it('sends a PUT request to the \'importAuth\' url passed in settings, calls updateFromData with the response, and calls activatePicker', (done) => {
+                vm.updateFromData(data)
+                    .always(function() {
+                        vm._importAuthConfirm()
+                            .always(function() {
+                                assert.isTrue(
```

Comment:
Since we expose the sinon.assert interface to chai, you can do `assert.calledWith(putJSONSpy, importAuthUrl, {}`